dds/DCPS/DataWriterImpl_T.h

```diff
@@ -433,6 +450,43 @@ DataWriterImpl_T
     Message_Block_Ptr mb;
     ACE_Message_Block* tmp_mb;
 
+    if (marshal_skip_serialize_) {
+      size_t effective_size = 0, padding = 0;
+      if (marshaled_size_) {
+        effective_size = marshaled_size_;
+      } else {
+        // TODO:  Fix this.
```

Comment:
@adamsj-oci This needs attention.

Reply:
@iguessthislldo suggests using `encoding_mode_.buffer_size(instance_data)`

Reply:
Are we saying
```  ACE_Message_Block* dds_marshal(const MessageType& instance_data,
                                 OpenDDS::DCPS::MarshalingType marshaling_type)
  {
    const bool encapsulated = cdr_encapsulation();
    const Encoding& encoding = encoding_mode_.encoding();
    Message_Block_Ptr mb;
    ACE_Message_Block* tmp_mb;

    if (marshal_skip_serialize_) {
      const size_t effective_size = encoding_mode_.buffer_size(instance_data);
      ACE_NEW_MALLOC_RETURN(tmp_mb,
        static_cast<ACE_Message_Block*>(
          mb_allocator_->malloc(sizeof(ACE_Message_Block))),
        ACE_Message_Block(
          effective_size,
          ACE_Message_Block::MB_DATA,
          0, // cont
          0, // data
          data_allocator_.get(), // allocator_strategy
          get_db_lock(), // data block locking_strategy
          ACE_DEFAULT_MESSAGE_BLOCK_PRIORITY,
          ACE_Time_Value::zero,
          ACE_Time_Value::max_time,
          db_allocator_.get(),
          mb_allocator_.get()),
          0);
      mb.reset(tmp_mb);
      if (!MarshalTraitsType::to_message_block(*mb, instance_data)) {
        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: DataReaderImpl::dds_demarshal: ")
                   ACE_TEXT("attempting to skip serialize but bad from_message_block.\n")));
      }
      return mb.release();
    }
```

Reply:
Yes I think. You can put `encoding_mode_.buffer_size(instance_data)` into the constructor like the other ones.

Reply:
That's what I was expecting to see based on Fred's suggestion. There is a test `SkipSerialize` checking this so if that passes, then I'm good.